cookbooks/custom_formatter.rst

```diff
@@ -0,0 +1,464 @@
+        */
+       private function doWrite(array $messages, bool $append): void
+       {
+           // create the output path if if does not exists.
+           if (!is_dir($this->getOutputPath())) {
+               mkdir($this->getOutputPath(), 0777, true);
+           }
+
+           // write data to the file
+           file_put_contents($this->getFilePath(), implode("\n", $messages) . "\n", $append ? \FILE_APPEND : 0);
+       }
+
+       private function getFilePath(): string
+       {
+           return $this->getOutputPath() . '/' . $this->fileName;
+       }
+   }
+
+Integration in your project
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You need to add the extension in you behat configuration file (default is ``behat.yml``) and configure it to use the formatter:
+
+.. code:: yaml
+
+   default:
+     extensions:
+       JDeniau\BehatReviewdogFormatter\ReviewdogFormatterExtension: ~
+       
+     formatters:
+       pretty: true
+       reviewdog: # "reviewdog" here is the "name" given in our formatter
+         # output_path is optional and handled directy by behat
+         output_path: 'build/logs/behat'
+         # file_name is optional and a custom parameter that we inject into the printer
+         file_name: 'reviewdog-behat.json'
+
+Different output per profile
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can active the extension only for a certain profile by specifying a profile in your command (ex: ``--profile=ci``)
+
+For example if you want the pretty formatter by default, but both progress and reviewdog on your CI, you can configure it like that:
+
+.. code:: yaml
+
+   default:
+     extensions:
+       JDeniau\BehatReviewdogFormatter\ReviewdogFormatterExtension: ~
+       
+     formatters:
+       pretty: true
+
+   ci:
+     formatters:
+       pretty: false
+       progress: true
+       reviewdog:
+         output_path: 'build/logs/behat'
+         file_name: 'reviewdog-behat.json'
+
+
+Enjoy !
+-------
+
+That's how you can write a simple custom behat formatter !
+
+If you have much more complex logic, and you need to formatter to be more dynamic, behat do provide a FormatterFactory interface.
+You can see usage examples directly in `behat's codebase <https://github.com/Behat/Behat/tree/2a3832d9cb853a794af3a576f9e524ae460f3340/src/Behat/Behat/Output/ServiceContainer/Formatter>`__,
+but in a lot of cases, the simple formatter should work.
+
+Want to use reviewdog and the custom formatter yourself ?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to use the reviewdog custom formatter, you can find it on github: https://github.com/jdeniau/behat-reviewdog-formatter
+
+There are other behat custom formatters in the wild, especially `BehatHtmlFormatterPlugin <https://github.com/dutchiexl/BehatHtmlFormatterPlugin>`__,
+that I did not test, but helped me understand how does behat formatter system works, and can output an HTML file that can help you understand why your CI is failing.
```

Comment:
```suggestion
There are other behat custom formatters in the wild, especially `BehatHtmlFormatterPlugin <https://github.com/dutchiexl/BehatHtmlFormatterPlugin>`__.
I did not test that, but it helped me understand how the behat formatter system works, and it can output an HTML file that can help you understand why your CI is failing.
```